kotlinx-coroutines-core/common/src/channels/ArrayChannel.kt

```diff
@@ -123,6 +130,16 @@ internal open class ArrayChannel<E>(
         return receive!!.offerResult
     }
 
+    // Guarded by lock
+    private fun ensureCapacity(currentSize: Int) {
+        if (currentSize == buffer.size) {
```

Comment:
Nitpick: It is a good style use `>=` in comparisons like this (aka "defensive programming"). Not that it really matters, but it makes code slightly easier to understand.